### PR TITLE
Core/Spells: Paladin auras + gift of the naaru + penance

### LINKED ISSUE #21227: Core/Spell: Penance
**Description:**
Penance damages the enemy and the player too.

**Expected behaviour:**
Penance shouldn't damage the player when targeting an enemy.

**Steps to reproduce the problem:**

1. Create a priest
2. Level up to 10
3. Choose Discipline spec
4. Hit an enemy with Penance

**Branch(es):** master

**TC rev. hash/commit:** https://github.com/TrinityCore/TrinityCore/commit/3d7af56f5cab503518fb2202ca839793c2b08862

**TDB version:**  720 + Updates

**Operating system:** Windows 10 x64

### LINKED ISSUE #22468: Gift of the naaru
**Current behaviour:**
Draenei Racial Spell Gift of the Naaru Heal 20 points of the Hp Bar

**Expected behaviour:**

Draenei Racial Spell Gift of the Naaru Heal 20 % of the Hp Bar.

**Branch(es):**
Master

**Operating system:** Win 10 Pro

### LINKED ISSUE #26125: Core/Spells Paladin multiple Auras can be active
**Description:**

Paladin can stack his Aura spells and activate them all.

Spells: 32223, 465, 317920, 183435

**Expected behaviour:**

There should be only one active Aura.
If you activate a new one, the old Aura gets disabled

**Steps to reproduce the problem:**

1. create a Paladin
2. set level to 60
3. use your Auraactionbar and activate your auras

**Branch(es):**

master

**TC rev. hash/commit:** 

01a3be276f4076762290e985d10d5c4ab1176ce0

**Operating system:** 

Win10

### PR DESCRIPTION
**Pull request by matanshukry**
**Changes proposed:**

-  Paladins auras: now only 1 can be active
-  Gift of the naaru now corerctly heals for 20% of the caster max health over 5 seconds
-  Gift of the naaru now also works for monk

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**

Closes #26125
Closes #22468
Closes #21227

**Tests performed:**

- [x] It builds
- [x] Tested in-game

**Known issues and TODO list:**

### CODE REVIEW

File: src/server/scripts/Spells/spell_paladin.cpp [hunk 1]
```diff
@@ -184,29 +216,29 @@ class spell_pal_blessing_of_faith : public SpellScript
 
     void HandleDummy(SpellEffIndex /*effIndex*/)
     {
-        if (Unit* unitTarget = GetHitUnit())
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_paladin.cpp:187]: indentation, also unrelated change i guess

File: src/server/scripts/Spells/spell_generic.cpp [hunk 2]
```diff
@@ -1322,28 +1329,7 @@ class spell_gen_gift_of_naaru : public AuraScript
...
-            case SPELLFAMILY_HUNTER:
-            case SPELLFAMILY_DEATHKNIGHT:
-                heal = 1.1f * float(std::max(GetCaster()->GetTotalAttackPowerValue(BASE_ATTACK), GetCaster()->GetTotalAttackPowerValue(RANGED_ATTACK)));
-                break;
-            case SPELLFAMILY_GENERIC:
-            default:
-                break;
-        }
-
+        float heal = 0.2f * GetCaster()->GetMaxHealth();
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_generic.cpp:1332]: we can get that 0.2f by reading basepoints of EFFECT1 / 100:
```
Effect 1: Id 3 (SPELL_EFFECT_DUMMY)
Difficulty: Id 0 (DIFFICULTY_NONE)
BasePoints = 20.00
Targets (0, 0) (NO_TARGET, NO_TARGET)
```

File: src/server/scripts/Spells/spell_paladin.cpp (hunk 1 above)
Comment (matanshukry) [reply src/server/scripts/Spells/spell_paladin.cpp:187]: yup, not sure why it even changed; don't remember touching that. Thanks

File: src/server/scripts/Spells/spell_generic.cpp [hunk 3]
```diff
@@ -1329,9 +1329,13 @@ class spell_gen_gift_of_naaru : public AuraScript
         if (!GetCaster())
             return;
 
-        float heal = 0.2f * GetCaster()->GetMaxHealth();
-        int32 healTick = std::floor(heal / aurEff->GetTotalTicks());
-        amount += int32(std::max(healTick, 0));
+        if (SpellEffectInfo const* eff1 = GetSpellInfo()->GetEffect(EFFECT_1))
+        {
+            float healPct = eff1->CalcValue() / 100.0f;
+            float heal = 0.2f * GetCaster()->GetMaxHealth();
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_generic.cpp:1335]: you forgot to replace 0.2f with healPct here :D

Comment (matanshukry) [reply src/server/scripts/Spells/spell_generic.cpp:1335]: lol just saw that too xd

File: src/server/scripts/Spells/spell_generic.cpp [hunk 4]
```diff
@@ -1322,30 +1329,13 @@ class spell_gen_gift_of_naaru : public AuraScript
...
-            case SPELLFAMILY_DEATHKNIGHT:
-                heal = 1.1f * float(std::max(GetCaster()->GetTotalAttackPowerValue(BASE_ATTACK), GetCaster()->GetTotalAttackPowerValue(RANGED_ATTACK)));
-                break;
-            case SPELLFAMILY_GENERIC:
-            default:
-                break;
+            float healPct = eff1->CalcValue() / 100.0f;
+            float heal = 0.2f * GetCaster()->GetMaxHealth();
+            int32 healTick = std::floor(heal / aurEff->GetTotalTicks());
+            amount += int32(std::max(healTick, 0));
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_generic.cpp:1337]: we may also remove that std::max here i guess?

Comment (matanshukry) [reply src/server/scripts/Spells/spell_generic.cpp:1337]: why?

Comment (mdX7) [reply src/server/scripts/Spells/spell_generic.cpp:1337]: healTick will always be >= 0

Comment (matanshukry) [reply src/server/scripts/Spells/spell_generic.cpp:1337]: Unless `eff->CalcValue()` is negative (changed in the future) ?

Comment (mdX7) [reply src/server/scripts/Spells/spell_generic.cpp:1337]: that'd require us to fix the spell either way (again) in that scenario, either it'll heal exactly 0 or it will deal damage (if < 0) so we might actually drop std::max for some minimal performance improvement

Comment (matanshukry) [reply src/server/scripts/Spells/spell_generic.cpp:1337]: cool, will do

**Comment by Shauren** [comment]
Auras: a9fa007f83f99ccf38bc60bb1a75f671db092781
Gift of the Naaru: d2b70397ab98fe8a24a5880b757a50e3736787d3
Penance: not pushed - it looks like ChannelObjects implementnation is not 100% correct in Spell class
